saleor/graphql/context.py

```diff
@@ -16,18 +17,23 @@ def get_context_value(request):
     return request
 
 
-def get_app(auth_token) -> Optional[App]:
-    tokens = AppToken.objects.filter(auth_token=auth_token).values("pk")
+def get_app(raw_auth_token) -> Optional[App]:
+    tokens = AppToken.objects.filter(token_last_4=raw_auth_token[-4:]).values_list(
+        "id", "auth_token"
+    )
+    token_ids = [
+        id for id, auth_token in tokens if check_password(raw_auth_token, auth_token)
+    ]
```

Comment:
I need to add iteration, as we right now do not have the raw auth token values saved, only the last 4 chars. IMO it shouldn't cause any performance problems as we iterate only through objects, in which the last 4 chars match the last chars of a given token.
But if anyone has a better idea, I'm open for any suggestions.

Reply:
We should be sure that this code is not executed before all requests from non-app users.